skopt/parameter.py

```diff
@@ -38,178 +37,156 @@ def inverse_transform(self, values):
         return 10**np.asarray(values)
 
 
-class Log(TransformerMixin):
-    """Natural logarithm transform."""
-    def fit(self, values):
-        return self
+class Distribution:
 
-    def transform(self, values):
-        return np.log(values)
+    @abc.abstractmethod
+    def rvs(self, n_samples=None, random_state=None):
+        """
+        Randomly sample points from the original space
+        """
+        return
 
-    def inverse_transform(self, values):
-        return np.exp(values)
+    def transform(self, random_vals):
+        """
+        Transform points to a warped space.
+        """
+        return self.transformer.transform(random_vals)
+
+    def inverse_transform(self, random_vals):
+        """
+        Transform points from a warped space.
+        """
+        return self.transformer.inverse_transform(random_vals)
 
 
-class CategoryTransform(TransformerMixin):
+class CategoricalEncoder(object):
+    """
+    OneHotEncoder of scikit-learn that can handle categorical
+    variables.
+    """
     def __init__(self):
         """Convert labeled categories into one-hot encoded features"""
         self._label = LabelEncoder()
         self._onehot = OneHotEncoder()
 
     def fit(self, values):
+        """
+        Fit a list or array of categories.
+
+        Parameters
+        ----------
+        * `values` [array-like]:
+            List of categories.
+        """
         vals = np.asarray(values)
         vals = self._label.fit_transform(vals)
         self._onehot.fit(vals.reshape(-1, 1))
-
         return self
 
     def transform(self, values):
-        vals = np.asarray(values)
-        vals = self._label.transform(vals).reshape(-1, 1)
-        return self._onehot.transform(vals).toarray()
+        """
+        Transform an array of categories to a one-hot encoded representation.
 
-    def inverse_transform(self, values):
```

Comment:
Do you know why it doesn't have `inverse_transform`? I can't think of a reason other than "hasn't been implemented yet".

How do we present the result of the optimisation to the user if there is no `inverse_transform`? I like @glouppe's idea of hiding the transform from the user, but in that case we need to `inverse_transform` :-/